sipp/src/lib.rs

```diff
@@ -62,31 +75,35 @@ impl<E: PairingEngine, D: Digest> SIPP<E, D> {
             let z_l = product_of_pairings::<E>(a_r, b_l);
             let z_r = product_of_pairings::<E>(a_l, b_r);
             proof_vec.push((z_l, z_r));
-            rng.absorb(&to_bytes![z_l, z_r].unwrap());
-            let x: E::Fr = u128::rand(&mut rng).into();
+            {
+                let mut buf = Vec::new();
+                (z_l, z_r).serialize_uncompressed(&mut buf).unwrap();
+                rng.absorb(&buf);
```

Comment:
We might want to make a `serialize_to_vec` as a drop-in replacement for `to_bytes` that essentially internally does this block.

Reply:
PR here https://github.com/arkworks-rs/algebra/pull/693